File: Snake_Java/src/snake/Poison.java
package snake;

import javax.swing.*;
import java.awt.*;

/**
 * Poison.java
 *
 * Contains attributes and methods for drawing
 * and displaying of poisons
 */
public class Poison extends Component implements Drawable
{
    private int xPos, yPos;
    private static final String imgPath ="images/poison.png";

    public Poison(int x, int y)
    {
        xPos = x;
        yPos = y;
    }

    /**
     * Draws an image of a poison
     *
     * @param g: Graphics object
     */
    public void draw(Graphics g)
    {
        ImageIcon img = new ImageIcon(imgPath);
        img.paintIcon(this, g, xPos, yPos);
    }

    /**
     * @return x position of a poison
     */
    public int getXPos()
    {
        return xPos;
    }

    /**
     * @param xPos:	x position to set
     */
    public void setXPos(int xPos)
    {
        this.xPos = xPos;
    }

    /**
     * @return y position of a poison
     */
    public int getYPos()
    {
        return yPos;
    }

    /**
     * @param yPos: y position to set
     */
    public void setYPos(int yPos)
    {
        this.yPos = yPos;
    }
}

File: Snake_C++/src/Engine/Poison.cpp
#include "../../include/Poison.hpp"

using namespace Engine;

Poison::Poison() = default;

Poison::Poison(std::shared_ptr<Context> &context, int xPos, int yPos)
      : context(context), xPos(xPos), yPos(yPos)
{
}

/**
 * Initialise poison image and draws it to the current position
 */
void Poison::draw()
{
    context->assets->addTexture(POISON, "../assets/images/poison_skull.png");
    poison.setTexture(context->assets->getTexture(POISON));
    poison.setPosition((float)xPos, (float)yPos);
    context->window->draw(poison);
}

/**
 * @return x position of a poison
 */
int Poison::getXPos()
{
    return xPos;
}

/**
 * @param x:	x position to set
 */
void Poison::setXPos(int x)
{
    this->xPos = x;
}

/**
 * @return y position of a poison
 */
int Poison::getYPos()
{
    return yPos;
}

/**
 * @param y: y position to set
 */
void Poison::setYPos(int y)
{
    this->yPos = y;
}
